app/services/bulk_tagging/build_tag_migration.rb

```diff
@@ -2,20 +2,16 @@ module BulkTagging
   class BuildTagMigration
     class InvalidArgumentError < ArgumentError; end
 
-    attr_reader :tag_migration_params, :taxon_content_ids, :content_base_paths
+    attr_reader :source_content_item, :taxon_content_ids, :content_base_paths
 
-    def initialize(tag_migration_params:, taxon_content_ids:, content_base_paths:)
-      @tag_migration_params = tag_migration_params
+    def initialize(source_content_item, taxon_content_ids, content_base_paths)
+      @source_content_item = source_content_item
       @taxon_content_ids = taxon_content_ids
       @content_base_paths = content_base_paths
     end
 
-    def self.call(tag_migration_params:, taxon_content_ids:, content_base_paths:)
-      new(
-        tag_migration_params: tag_migration_params,
-        taxon_content_ids: taxon_content_ids,
-        content_base_paths: content_base_paths
-      ).call
+    def self.call(source_content_item:, taxon_content_ids:, content_base_paths:)
```

Comment:
I agree - the pattern makes it hard to refactor. I've kept it to keep it consistent. Changing it can be done in another PR since I'd like to discuss it with @carvil / @mobaig first.